Onboarding dbt-fabricspark adapter, adding documentation of the adapter (#7219)
dbt-labs/docs.getdbt.com: website/docs/reference/resource-configs/fabricspark-configs.md

---
title: "Microsoft Fabric Spark configurations"
id: "fabricspark-configs"
description: "Microsoft Fabric Spark Configurations - Read this in-depth guide to learn about configurations in dbt."
---

## Configuring tables

When materializing a model as `table`, you may include several optional configs that are specific to the dbt-spark plugin, in addition to the standard [model configs](/reference/model-configs).

| Option  | Description          | Required?        | <div style={{width:'350px'}}>Example</div>       |
|---------|----------------------|------------------|--------------------------------------------------|

> | file_format | The file format to use when creating tables (`parquet`, `delta`, `csv`). | Optional | `delta`|

Reviewer: Did this get carried from dbt-spark? because delta is not a file format?

Author: yes, and it is applicable to fabric spark adapter as well. the fabric spark adapter uses file_format when creating tables, same as spark. Delta is a table format but using the same property name (file_format) for ease.

Author: Providing the file_format value is optional because all tables are created as Delta by default. However, some materializations do not require the table to be in Delta format, which is why specifying it is not mandatory.

Reviewer: ah okay - it's been on my list to go fix this (it should be two separate fields).

> | location_root [^1]  | The created table uses the specified directory to store its data. The table alias is appended to it.                               | Optional                | `Files/<folder>` or `Tables/<tableName>`              |

Reviewer: This is a very short footnote. I think you should just put it in this box

> | location_root [^1]  | The created table uses the specified directory to store its data. The table alias is appended to it.                               | Optional                | `Files/<folder>` or `Tables/<tableName>`              |

Reviewer: ```suggestion
| location_root [^1]  | The specified directory used to store table data. The table alias is appended to it.                               | Optional                | `Files/<folder>` or `Tables/<tableName>`              |
```

> | partition_by  | Partition the created table by the specified columns. A directory is created for each partition.                                   | Optional                | `date_day`              |

Reviewer: ```suggestion
| partition_by  | Partition the table by the specified columns. A directory is created for each partition.                                   | Optional                | `date_day`              |
```

> | clustered_by  | Each partition in the created table will be split into a fixed number of buckets by the specified columns.                         | Optional               | `country_code`              |

Reviewer: ```suggestion
| clustered_by  | Each partition in the table will be split into a fixed number of buckets by the specified columns.                         | Optional               | `country_code`              |
```

| buckets  | The number of buckets to create while clustering                                                                                   | Required if `clustered_by` is specified                | `8`              |

> | tblproperties | The table properties configure table behavior. Properties differ depending on the file format, see reference docs ([Iceberg](https://iceberg.apache.org/docs/latest/configuration/#table-properties), [Parquet](https://spark.apache.org/docs/3.5.4/sql-data-sources-parquet.html#data-source-option), [Delta](https://docs.databricks.com/aws/en/delta/table-properties#delta-table-properties), [Hudi](https://hudi.apache.org/docs/sql_ddl/#table-properties)). | Optional |<code>Provider=delta Location=abfss://.../Files/tables/sales_data TableProperty.created.by=data_engineering_team TableProperty.purpose=sales analytics CreatedBy=Delta Lake CreatedAt=2024-12-01 14:21:00 Format=Parquet PartitionColumns=region MinReaderVersion=1 MinWriterVersion=2</code> |

Reviewer: what is the difference between file_format vs tblproperties?

Author: file_format drives the storage format of the table.

tblproperties provide information about the table like the location, created by, created at, column information and delta min reader and write protocol versions, format and partitions

| Option  | Description          | Required?        | <div style={{width:'350px'}}>Example</div>       |
|---------|----------------------|------------------|--------------------------------------------------|
| file_format | The file format to use when creating tables (`parquet`, `delta`, `csv`). | Optional | `delta`|
| location_root [^1]  | The created table uses the specified directory to store its data. The table alias is appended to it.                               | Optional                | `Files/<folder>` or `Tables/<tableName>`              |
| partition_by  | Partition the created table by the specified columns. A directory is created for each partition.                                   | Optional                | `date_day`              |
| clustered_by  | Each partition in the created table will be split into a fixed number of buckets by the specified columns.                         | Optional               | `country_code`              |
| buckets  | The number of buckets to create while clustering                                                                                   | Required if `clustered_by` is specified                | `8`              |

> | tblproperties | The table properties configure table behavior. Properties differ depending on the file format, see reference docs ([Iceberg](https://iceberg.apache.org/docs/latest/configuration/#table-properties), [Parquet](https://spark.apache.org/docs/3.5.4/sql-data-sources-parquet.html#data-source-option), [Delta](https://docs.databricks.com/aws/en/delta/table-properties#delta-table-properties), [Hudi](https://hudi.apache.org/docs/sql_ddl/#table-properties)). | Optional |<code>Provider=delta Location=abfss://.../Files/tables/sales_data TableProperty.created.by=data_engineering_team TableProperty.purpose=sales analytics CreatedBy=Delta Lake CreatedAt=2024-12-01 14:21:00 Format=Parquet PartitionColumns=region MinReaderVersion=1 MinWriterVersion=2

Reviewer 2: ```suggestion
| tblproperties | The table properties configure table behavior. Properties differ depending on the file format, see reference docs ([Iceberg](https://iceberg.apache.org/docs/latest/configuration/#table-properties), [Parquet](https://spark.apache.org/docs/3.5.4/sql-data-sources-parquet.html#data-source-option), [Delta](https://docs.databricks.com/aws/en/delta/table-properties#delta-table-properties), [Hudi](https://hudi.apache.org/docs/sql_ddl/#table-properties)). | Optional |<code>Provider=delta Location=abfss://.../Files/tables/sales_data TableProperty.created.by=data_engineering_team TableProperty.purpose=sales analytics CreatedBy=Delta Lake CreatedAt=2024-12-01 14:21:00 Format=Parquet PartitionColumns=region MinReaderVersion=1 MinWriterVersion=2</code> |
```

> </code> |

Reviewer 2: ```suggestion
```
